docs/concepts/juttle_tutorial.md

```diff
@@ -2,27 +2,1055 @@
-documentation than this, see [Juttle concepts](juttle_overview.html). 
+This tutorial works best if you install the juttle-engine assembly that provides both the CLI and the browser execution environment. This can be done via `npm install juttle-engine`, or using a docker container we provide. The demo system demo.juttle.io is a public deployment of that container.
 
-In Juttle, *the basic unit of data is a point*. A point consists of a
-number of key/value pairs, where the keys are strings and the values are
-numbers, strings, Booleans, or null. Points flow through flowgraphs and
-can be transformed, aggregated, or joined at each processing step.
+### Demo Option
 
+You can follow along by executing the Juttle programs, and viewing their source code, on the demo system at URL:
+
+demo.juttle.io/?path=/examples/github-tutorial/index.html
+
+The [RUN] links throughout this tutorial take you to execute a given program on demo.juttle.io.
+
+This will give you the browser experience via juttle-viewer app, but not a CLI. It works as a quick ride-along option; to get the most out of the tutorial, you'll want to have a local installation of the Juttle Engine.
+
+### Local Option
+
+Install the juttle-engine assembly with:
+```
+npm install -g juttle-engine
+```
+If you did a global install with `-g` as above, that makes `juttle` CLI binary available anywhere:
+```
+$ juttle
+juttle>
+```
+
+We will be using Elasticsearch as a data store for part 2 of the tutorial; you can run it locally or in a docker container. You will need to configure the elastic adapter section in your `~/.juttle/config.json` file.
+
+The Juttle programs used in this tutorial are available under your `${NODE_MODULES_DIR}/juttle-engine/examples/github-tutorial/juttles/`. Let's copy them to an easily accessible location:
+
+```
+mkdir -p /examples/tutorial; cp ${NODE_MODULES_DIR}/juttle-engine/examples/github-tutorial/juttles/* /examples/tutorial/
+```
+
+### Dockerized Option
+
+Follow instructions in [juttle-engine tutorial example](https://github.com/juttle/juttle-engine/tree/master/examples/github-tutorial/README.md), tl;dr is
+
+```
+git clone https://github.com/juttle/juttle-engine
+cd juttle-engine/examples && docker-compose -f dc-juttle-engine.yml -f github-tutorial/dc-elastic.yml up -d
+DOCKER_MACHINE_HOST=`docker-machine ip default` && open http://$DOCKER_MACHINE_HOST:8080/?path=/examples/tutorial/index.juttle
+```
+
+Your browser will open the page linking to juttle programs with visual output (the programs are included in the docker container). You can run all of the examples in this fashion, but if you want to modify code as you try it out, best start with the CLI.
+
+To launch the Juttle CLI in interactive mode, you would do:
+
+```
+docker exec -it examples_juttle-engine_1 juttle
+```
+
+To run programs via CLI from saved juttle files, you would do:
+
+```
+docker exec examples_juttle-engine_1 juttle /examples/tutorial/filename.juttle
```

Comment:
I think the path within the container is /opt/juttle-engine/juttles/examples, so when running juttle you need to specify that path, not the path relative to the juttle-engine root.